src/NLPModelsIpopt.jl

```diff
@@ -181,6 +182,41 @@ function ipopt(nlp::AbstractNLPModel; kwargs...)
   return solve!(solver, nlp, stats; kwargs...)
 end
 
+"""
+    ipopt(nls::AbstractNLSModel; kwargs...)
+
+Solve the least-squares problem `nls` using `IPOPT` by moving the nonlinear residual to the constraints.
+
+# Arguments
+- `nls::AbstractNLSModel`: The least-squares problem to solve.
+
+For advanced usage, first define an `IpoptSolver` to preallocate the memory used in the algorithm, and then call `solve!`:
+    solver = IpoptSolver(nls)
+    solve!(solver, nls; kwargs...)
+
+# Examples
+```julia
+using NLPModelsIpopt, ADNLPModels
+nls = ADNLSModel(x -> [x[1] - 1, x[2] - 2], [0.0, 0.0], 2)
+stats = ipopt(nls, print_level = 0)
+```
+"""
+function ipopt(ff_nls::FeasibilityFormNLS; kwargs...)
+  solver = IpoptSolver(ff_nls)
+  stats = GenericExecutionStats(ff_nls)
+  stats = solve!(solver, ff_nls, stats; kwargs...)
+
+  return stats
+end
+
+function ipopt(nls::AbstractNLSModel; kwargs...)
+  ff_nls = FeasibilityFormNLS(nls)
+  stats = ipopt(ff_nls; kwargs...)
+  # Only keep the original variables in the solution
+  stats.solution = stats.solution[1:nls.meta.nvar]
```

Comment:
Following this https://github.com/JuliaSmoothOptimizers/SolverCore.jl/blob/main/src/stats.jl ,
we also need to change stats.multipliers_L, stats.multipliers_U and stats.multipliers. The first two are of the size `nvar` while the last is of the size `ncon`, so we extract the last `nls.meta.ncon` entries https://github.com/JuliaSmoothOptimizers/NLPModelsModifiers.jl/blob/40655dd6fc140da39da4cf630062d7b0ebf54c05/src/feasibility-form-nls.jl#L40.